### PR TITLE
WIP: Workflow

### PR DESCRIPTION
**Pull request by hiaselhans**
With your PR, here is a check list:

- [ ] Has test cases written?
- [ ] Has all code lines tested?
- [ ] Has `make format` been run?
- [ ] Please update CHANGELOG.yml(not CHANGELOG.rst)
- [ ] Passes all Travis CI builds
- [ ] Has fair amount of documentation if your change is complex
- [ ] Agree on NEW BSD License for your contribution

Don't know if you're fond of this, but this uses github's actions for unittests and linting.
the thing is that with different builds some tests fail. Same as on my local system (linux 5.9.2, python 3.8.6)
With all systems and python versions used, plus builds on forked repos we'd have a cleaner experience for contributors

### CODE REVIEW

File: .github/workflows/moban-update.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
-on: [push]
+on: 
+  push:
+    branches:
+      - main, dev
```
Comment (chfw) [inline .github/workflows/moban-update.yml:4]: pyexcel still has the old style 'master' branch as release branch

File: .github/workflows/moban-update.yml [hunk 2]
```diff
@@ -1,4 +1,7 @@
-on: [push]
```
Comment (chfw) [inline .github/workflows/moban-update.yml:1]: this file itself, would be updated by moban, hence the proper place update it is: https://github.com/moremoban/pypi-mobans/blob/dev/statics/moban-update.yml

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: run_tests
+
+on: [push, pull_request]
+
+jobs:
```
Comment (chfw) [inline .github/workflows/tests.yml:5]: your work here resolves issue on this project: https://github.com/moremoban/pypi-mobans/issues/148

which meant: all projects using pypi-mobans would benefit from your work, including pyexcel family projects.

File: .github/workflows/moban-update.yml (hunk 2 above)
Comment (hiaselhans) [reply .github/workflows/moban-update.yml:1]: ok sure! will do that.
Reason for this being that commits were pushed to forks and pr's

Comment (hiaselhans) [reply .github/workflows/moban-update.yml:1]: Ok i don't completely get it.
moban is used to create those files?
How is moban supposed to know the master branch // master repo name?

File: .github/workflows/tests.yml (hunk 3 above)
Comment (hiaselhans) [reply .github/workflows/tests.yml:5]: cool! :)

and yes, it's as simple as that:
```
matrix:
    python-version: [3.6, 3.7, 3.8, 3.9]
    os: [macOs-latest, ubuntu-latest, windows-latest]
```

File: .github/workflows/moban-update.yml (hunk 2 above)
Comment (chfw) [reply .github/workflows/moban-update.yml:1]: .moban.yml has the answers.

Comment (chfw) [reply .github/workflows/moban-update.yml:1]: https://moban.readthedocs.io/en/latest/ Has the long answer

Comment (chfw) [reply .github/workflows/moban-update.yml:1]: how it works with pyexcel is: 

a template/source file A.jj2 can be stored in either of the following two:

1. pypi-mobans, generic template source
2. pyexcel-mobans, pyexcel specific source

.moban.yml in pyexcel would have:

targets:
  - my_local_A: A.jj2

And to make it a bit complex is:

pyexcel-mobans had a [mobanfile](https://github.com/pyexcel/pyexcel-mobans/blob/master/mobanfile.yaml), where common sections among pyexcel family are stored.

so each individual project only store their specific targets. 

In moban-update.yml case, it is mentioned in https://github.com/pyexcel/pyexcel-mobans/blob/master/mobanfile.yaml#L20 . so if your update is reviewed and committed in pypi-mobans project, moban command read .moban.yml and bring your update forward to pyexcel project.

here is the idea about all above fluffy admin stuff: https://moban.readthedocs.io/en/latest/#at-scale-continous-templating-for-open-source-projects

Comment (hiaselhans) [reply .github/workflows/moban-update.yml]: Ok thanks for explination.

i think it'd take me a long way to look up all the internals so I removed that part of the pr.

Main issue being that tests fail on windows (file conflict) and doctests failing in python > 3.6...

Comment (hiaselhans) [reply .github/workflows/moban-update.yml]: and the main intention is that workflows are better integrated and much faster

**Comment by chfw** [comment]
will merge this PR after 0.6.6 release. I see that the failures are down to doc tests.

please expect that the files will be relocated to template projects, from which the file will come back.

I expect that after the merge, some files in this PR will be overwritten.

**Comment by hiaselhans** [comment]
thx @chfw !

**Pull request merged**